proposals/VK_EXT_present_timing.adoc

```diff
@@ -0,0 +1,470 @@
+
+                if (timings[i].timeDomainChanged || timings[i].timingPropertiesChanged) {
+                    // Re-query time domains / timing properties, invalidate results
+                    // (...)
+                }
+            }
+        }
+
+        // Process past presentation timings:
+        // - Statistically determine whether changing the IPD is necessary (or desired).
+        // - Based on previous reported times and target IPD, compute the next target present time.
+        VkPresentTimeEXT targetPresentTime = ProcessPastPresentationTimings(pastPresentationTimings);
+
+        // Position scene geometry / camera for `targetPresentTime'
+        // (...)
+
+        vkAcquireNextImageKHR(...);
+
+        // Render to acquired swapchain image
+        // (...)
+
+        VkPresentTimesInfoEXT presentTimesInfo = {
+            .sType = VK_STRUCTURE_TYPE_PRESENT_TIMES_INFO_EXT,
+            .pNext = NULL,
+            .swapchainCount = 1,
+            .pTimes = &targetPresentTime
+        };
+
+        VkPresentIdKHR presentId = {
+            .sType = VK_STRUCTURE_TYPE_PRESENT_ID_KHR,
+            .pNext = &presentTimesInfo,
+            .swapchainCount = 1,
+            .pPresentIds = &currentPresentId
+        }
+
+        VkPresentInfoKHR presentInfo = {
+            .sType = VK_STRUCTURE_TYPE_PRESENT_INFO_KHR,
+            .pNext = &presentId,
+            // (...)
+        };
+
+        vkQueuePresentKHR(queue, &presentInfo);
+
+        currentPresentId++;
+    }
+----
+
+== Issues
+
+=== What are the key differences to `VK_GOOGLE_display_timing`?
+
+The API changes from `VK_GOOGLE_display_timing` are:
+
+* Introduction of present stages with `VkPresentStageFlagsEXT`
+* Rely on `VK_KHR_present_id` to specify present Ids
+* Expose features and present stages in physical device and surface features
+* Variable refresh rate indicator
+* Progressive timings feedback
+* Use a flag instead of a present slop to round to the nearest refresh cycle
+
+Compared to `VK_GOOGLE_display_timing`, stricter specification language is also used to allow for more consistent and wider adoption among implementors.
+
+=== RESOLVED: How does the application choose the value for `VkSwapchainPresentTimingCreateInfoEXT::presentTimingQueueSize`?
+
+Use reasonable default values, such as a multiple of the swapchain image count.
+
+Because presenting when the swapchain's internal timing queue is full is considered an error, the latency of the timing results effectively can end up throttling the present rate if the internal queue is small enough. The topology of the presentation engine usually being generally opaque to applications, there is no indication of the feedback latency before the application starts presenting.
+
+Applications which run into feedback latency issues may have to recreate their swapchains to resize the internal timing queue.
```

Comment:
It will be better to allow the queue size to be adjusted for a swapchain, without having to recreate the swapchain (which will cause visual anomalies).